### ISSUE TITLE
docker service not starting on ubuntu 14.04

### ISSUE THREAD

**Issue by tonnydourado** [garethr/garethr-docker#415]
The installation works, but puppet can't run the service. Running `service docker start` afterwards works, though. Here's the output:

````
Warning: Setting templatedir is deprecated. See http://links.puppetlabs.com/env-settings-deprecations
   (at /usr/lib/ruby/vendor_ruby/puppet/settings.rb:1139:in `issue_deprecation_warning')
Info: Loading facts
Info: Loading facts
Warning: Config file /etc/puppet/hiera.yaml not found, using Hiera defaults
Warning: Scope(Apt::Source[docker]): $include_src is deprecated and will be removed in the next major release, please use $include => { 'src' => false } instead
Warning: Scope(Apt::Source[docker]): $required_packages is deprecated and will be removed in the next major release, please use package resources instead.
Warning: Scope(Apt::Source[docker]): $key_source is deprecated and will be removed in the next major release, please use $key => { 'source' => https://apt.dockerproject.org/gpg } instead.
Warning: Scope(Apt::Key[Add key: 58118E89F3A912897C070ADBF76221572C52609D from Apt::Source docker]): $key_source is deprecated and will be removed in the next major release. Please use $source instead.
Notice: Compiled catalog for poamotbtdc014.home in environment production in 0.57 seconds
Info: Applying configuration version '1453676294'
Error: /Stage[main]/Docker::Service/Service[docker]: Could not evaluate: Execution of '/sbin/status docker' returned 1: status: Unknown job: docker
Notice: Finished catalog run in 1.08 seconds
````

And here's how I'm using it:

````
class {"docker":
    docker_users => ["mysuser"]
}

````

**Comment by tonnydourado** [comment]
I did a test and the problem seems to be that I'm running puppet with sudo -E, which I needed to be able to pass facts as environment variables (like FACTER_stuff="stuff"). Seems like /sbin/status looks at something in the environment variables to decide if it's gonna work properly or not, and when I pass my user environment (with the facter env vars), it doesn't work.

I figured out another way to pass the environment variables (I was using python and just passing os.environ to the subprocess, but now I'm embedding the env vars in the command itself, after the sudo), so, if you don't there's anything worth fixing, I guess this can be closed.
